src/stores/SpaceStore.tsx

```diff
@@ -366,16 +366,18 @@ export class SpaceStoreClass extends AsyncStoreWithClient<IState> {
     }
 
     public getParents(roomId: string, canonicalOnly = false): Room[] {
+        const userId = this.matrixClient?.getUserId();
         const room = this.matrixClient?.getRoom(roomId);
         return room?.currentState.getStateEvents(EventType.SpaceParent)
-            .filter(ev => {
+            .map(ev => {
                 const content = ev.getContent();
-                if (!content?.via?.length) return false;
-                // TODO apply permissions check to verify that the parent mapping is valid
-                if (canonicalOnly && !content?.canonical) return false;
-                return true;
+                if (!Array.isArray(content?.via)) return;
```

Comment:
Suggestion: I find the `return;` here a bit hard to grok initially - this is a function that returns values, so I'd expect to see an explicit `return undefined`. Just at a glance it looks like the `returns` are strictly for bailing out rather than returning something that's going to be filtered on.